gen.go

```diff
@@ -161,6 +161,12 @@ import (%s)
 func (g *generator) checkType(typName string) error {
 	var check func(t types.Type, ctx string) error
 	check = func(t types.Type, ctx string) error {
+		// If the type already implements both the marshal and unmarshal interface
+		// we can skip checking since the generater will use them.
+		if types.Implements(t, siaMarshaler) && types.Implements(types.NewPointer(t), siaUnmarshaler) {
```

Comment:
Huh. Initially I thought that this should be:
```go
if (types.Implements(t, siaMarshaler) || types.Implements(types.NewPointer(t), siaMarshaler)) &&
   types.Implements(types.NewPointer(t), siaUnmarshaler) {
```
but I think `types.Implements` might [already do this automatically](https://github.com/golang/go/blob/109d7580a5c507b1e4f460445a5c4cd7313e4aa6/src/go/types/lookup.go#L358). This matches Go's interface implementation behavior, so I guess I shouldn't be *too* surprised...